pkg/activitypub/store/ariesstore/ariesstore.go

```diff
@@ -0,0 +1,453 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package ariesstore
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"net/url"
+	"strconv"
+	"time"
+
+	ariesstorage "github.com/hyperledger/aries-framework-go/spi/storage"
+	"github.com/trustbloc/edge-core/pkg/log"
+
+	"github.com/trustbloc/orb/pkg/activitypub/store/memstore"
+	"github.com/trustbloc/orb/pkg/activitypub/store/spi"
+	"github.com/trustbloc/orb/pkg/activitypub/store/storeutil"
+	"github.com/trustbloc/orb/pkg/activitypub/vocab"
+)
+
+const (
+	activityTag      = "Activity"
+	objectIRITagName = "ObjectIRI"
+	timeAddedTagName = "TimeAdded"
+)
+
+var logger = log.New("activitypub_store")
+
+// Provider implements an ActivityPub store backed by an Aries storage provider.
+type Provider struct {
+	serviceName     string
+	activityStore   ariesstorage.Store
+	referenceStores map[spi.ReferenceType]ariesstorage.Store
+	actorStore      ariesstorage.Store
+}
+
+// New returns a new ActivityPub storage provider.
+func New(provider ariesstorage.Provider, serviceName string) (*Provider, error) {
+	stores, err := openStores(provider)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open stores: %w", err)
+	}
+
+	return &Provider{
+		serviceName:     serviceName,
+		activityStore:   stores.activities,
+		referenceStores: stores.reference,
+		actorStore:      stores.actor,
+	}, nil
+}
+
+// PutActor stores the given actor.
+func (s *Provider) PutActor(actor *vocab.ActorType) error {
+	logger.Debugf("[%s] Storing actor [%s]", s.serviceName, actor.ID())
+
+	actorBytes, err := actor.MarshalJSON()
```

Comment:
Not sure - is it proper to call MarshalJSON directly or should json.Marshal be used? i.e. is there any difference?

Reply:
Looks like the end result is the same (same methods get called and the bytes are identical), but if `actor` is nil, then `actor.MarshalJSON()` will panic, while json.Marshal won't... so it seems to me that json.Marshal would actually be a better choice. I'll switch it over (and other places that call MarshalJSON)

Reply:
(also updated the unmarshal calls to use json.unmarshal)

Reply:
@bstasyszyn Speaking of panics due to nil values, with the current log statements I have, they will panic if you pass in a nil actor or activity. Which of these do you think is a better solution?

1. Check for nil and return error
2. Change interface to not accept pointers

Reply:
I don't know if we need to check for nils. I think that if the caller passes a nil activity then it's a bug.